### PR TITLE
feat(elasticsearch): Updates to elasticsearch configuration, dao, tests

### PR DESCRIPTION
**Pull request by david-leifker**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [x] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Review by jjoyce0510 — APPROVED** [review]
Reviewed Internally. Looks good to me.

File: metadata-ingestion/src/datahub/cli/docker_cli.py [hunk 1]
```diff
@@ -17,6 +17,7 @@
 import pydantic
 import requests
 from expandvars import expandvars
+from requests_file import FileAdapter
```
Comment (david-leifker) [inline metadata-ingestion/src/datahub/cli/docker_cli.py:20]: Note that intent here is to exercise the same logic that fetches compose files from github to be able to use file:// in the same way. There is logic in this class that is generally useful for running the smoke test locally.

File: metadata-jobs/mae-consumer/src/main/java/com/linkedin/metadata/kafka/hook/UpdateIndicesHook.java [hunk 2]
```diff
@@ -225,7 +242,44 @@ private void updateGraphService(Urn urn, AspectSpec aspectSpec, RecordTemplate a
         _graphService.removeEdgesFromNode(entry.getKey(), new ArrayList<>(entry.getValue()),
             newRelationshipFilter(new Filter().setOr(new ConjunctiveCriterionArray()), RelationshipDirection.OUTGOING));
       }
-      edgesToAdd.forEach(edge -> _graphService.addEdge(edge));
+      edgesToAdd.forEach(_graphService::addEdge);
+    }
+  }
+
+  // Todo
```
Comment (RyanHolstien) [inline metadata-jobs/mae-consumer/src/main/java/com/linkedin/metadata/kafka/hook/UpdateIndicesHook.java:249]: What's left to do here?

Comment (david-leifker) [reply metadata-jobs/mae-consumer/src/main/java/com/linkedin/metadata/kafka/hook/UpdateIndicesHook.java:249]: Nothing that I can see, will remove

File: metadata-service/factories/src/main/java/com/linkedin/gms/factory/search/ElasticSearchBulkProcessorFactory.java [hunk 3]
```diff
@@ -38,15 +38,22 @@ public class ElasticSearchBulkProcessorFactory {
   @Value("${elasticsearch.bulkProcessor.retryInterval}")
   private Long retryInterval;
 
+  @Value("#{new Boolean('${elasticsearch.bulkProcessor.async}')}")
```
[the remark this answers is not in the document]
Comment (david-leifker) [reply metadata-service/factories/src/main/java/com/linkedin/gms/factory/search/ElasticSearchBulkProcessorFactory.java:41]: I copied this from somewhere else in the code base. At one point I tried a few alternatives but they didn't seem to work so sticking with convention for this.

**Review by RyanHolstien — APPROVED** [review]

File: smoke-test/requests_wrapper/utils_requests_wrapper.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import requests
```
[the remark this answers is not in the document]
Comment (jjoyce0510) [reply smoke-test/requests_wrapper/utils_requests_wrapper.py:1]: Love this!

**Review by jjoyce0510 — APPROVED** [review]
LGTM. This is a big PR so let's watch OSS closely after merging.

Great work, David!

**Pull request merged**